src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -1245,8 +1245,12 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
         llvm::Value* const_list = builder->CreateAlloca(const_list_type, nullptr, "const_list");
         list_api->list_init(type_code, const_list, *module, x.n_args, x.n_args);
         int64_t ptr_loads_copy = ptr_loads;
-        ptr_loads = 1;
         for( size_t i = 0; i < x.n_args; i++ ) {
+            if (ASR::is_a<ASR::CPtr_t>(*ASRUtils::expr_type(x.m_args[i]))) {
+                ptr_loads = 0;
+            } else {
+                ptr_loads = 1;
+            }
```

Comment:
I think the issue here is that if `x.m_args[i]` is an argument, then we want `ptr_loads=0`, but if it is `pi` (as in the other example), where it is a local (or global) variable, we want `ptr_loads = 1`.